src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -532,16 +532,30 @@ public bool Remove(
             if (index < newSize)
             {
                 // We're removing an element from the middle of the heap.
-                // Pop the last element in the collection and sift downward from the removed index.
+                // Pop the last element in the collection and sift from the removed index.
                 (TElement Element, TPriority Priority) lastNode = nodes[newSize];
 
                 if (_comparer == null)
                 {
-                    MoveDownDefaultComparer(lastNode, index);
+                    if (Comparer<TPriority>.Default.Compare(lastNode.Priority, priority) < 0)
```

Comment:
In theory, we would not need to do anything in the case  `Compare` returning zero.

Reply:
We still need to move the last node in the position of removed one, which is handled by the workhorse method.

Reply:
Oh, I missed it wasn't moved yet. Sorry, my bad.